ihontaryk/homework_8/restaurant.py

```diff
@@ -0,0 +1,78 @@
+from collections import OrderedDict
+from functools import partialmethod
+
+
+class Restaurant:
+    menu = {'first_course': 'First Course',
+            'second_course': 'Second Course',
+            'dessert': 'Dessert',
+            'beverage': 'Beverage'}
+
+    def __init__(self):
+        self.order = OrderedDict()
+
+    def _add_item(self, key, count, price):
+        if key not in self.menu:
+            raise ValueError(f'{key} is not available in the menu')
+
+        name = self.menu[key]
+
+        if name not in self.order:
+            self.order[name] = 0
+        self.order[name] += count * price
+
+    _add_first_course = partialmethod(_add_item, "first_course")
+    _add_second_course = partialmethod(_add_item, "second_course")
+    _add_dessert = partialmethod(_add_item, "dessert")
+    _add_beverage = partialmethod(_add_item, "beverage")
+
+    def get_client_order(self, first, second, desert, beverages):
```

Comment:
bad approach where you need to exptend params based on amount of dishes, 
if you have here 100 kinds?

Reply:
I would suggest order class, also possible variant to make params - named only parameters

```
    def get_client_order(self, *, first, second, desert, beverages):

```